sympy/ntheory/residue_ntheory.py

```diff
@@ -615,16 +615,62 @@ def is_nthpow_residue(a, n, m):
 
     P. Hackman "Elementary Number Theory" (2009),  page 76
     """
-    if( primitive_root(m) == None ):
-        raise NotImplementedError("Not implemented for m which do not have any primitive root")
+    if n == 0:
+        if m == 1:
+            return False
+        return a == 1
     if n == 1:
         return True
     if n == 2:
         return is_quad_residue(a, m)
+    if primitive_root(m) == None:
+        x = factorint(m)
+        for prime, power in x.items():
+                if not _nthroot_mod_prime_power(a, n, prime, power):
+                    return False
+        return True
     f = totient(m)
     k = f // igcd(f, n)
     return pow(a, k, m) == 1
 
+
+def _nthroot_mod_prime_power(a, n, p, k):
+    """Returns true if solution for ``x**n == a (mod(p**k))`` exists."""
+    if (a % p) != 0:
+        if p == 2:
+            if k == 1:
+                return True
+            elif k == 2:
+                if not n & 1 and a % 4 == 3:
+                    return False
+                return True
+            c = 0
```

Comment:
in my code I use `trailing(n)` for this...I *thought* there was a way to do that but didn't remember til later the name of the function.